src/main/java/pl/jablonskanycz/bakery/clients/FileBasedClientRepository.java

```diff
@@ -1,30 +1,144 @@
 import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static java.nio.file.StandardOpenOption.APPEND;
 
 public class FileBasedClientRepository implements ClientRepository {
+    private static final Path clientPath = Path.of("src", "main", "resources", "CLIENT.csv");
+    private AddressRepository addressRepository = new FileBasedAddressRepository();
     @Override
     public List<Client> getAll() {
-        return null;
+            List<Client> clients = null;
+            try {
+                clients = Files.lines(clientPath)
+                .skip(1)
+                .map(line -> {
+                    String[] client = line.split(",");
+                    return new Client(
+                        Integer.parseInt(client[0]),
+                        client[1],
+                        client[2],
+                        addressRepository.findById(Integer.parseInt(client[3])));
+                })
+                .collect(Collectors.toList());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            return clients;
     }
 
     @Override
     public Client findBySurname(String surname) {
-        return null;
+        Client client = null;
+        try {
+            return Files.lines(clientPath)
+                    .skip(1)
+                    .map(line -> {
+                        String[] strings = line.split(",");
+                        return new Client(
+                                Integer.parseInt(strings[0]),
+                                strings[1],
+                                strings[2],
+                                addressRepository.findById(Integer.parseInt(strings[3])));
+                    })
+                    .filter(a -> surname.equals(a.getSurname()))
+                    .findFirst()
+                    .orElseThrow(() -> new NoSuchElementException("Client list is empty"));
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return client;
     }
 
     @Override
     public void addClient(Client clientToAdd) {
-
+        List<String> clientsList = null;
+        try {
+            clientsList = Files.lines(clientPath)
+                    .toList();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        String lastId = clientsList.get(clientsList.size() - 1).split(",")[0];
+        int newId = Integer.parseInt(lastId) + 1;
+        String line = "\n" + newId + "," + clientToAdd.getName() + "," + clientToAdd.getSurname() + "," + clientToAdd.getAddress();
+        try {
+            Files.writeString(clientPath, line, APPEND);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
     }
 
     @Override
-    public void updateClient(Client clientToUpdate) {
+    public void updateClient(Client clientWithOldData, Client clientWithNewData) {
```

Comment:
well done! 🎉 I know it was quite hard. 
Are you ready to take it to the next level :) ? 

Let's use new Java11 features, like `Files.readAllLines` or `Files.write` :)
 I'll leave this hint here for you to investigate it further but in case you would need more clues - please let me know and I'll provide them :)